### PR TITLE
feat(dinomaly): Add context-aware recentering option from dinomaly2

### PR DESCRIPTION
**Pull request by rajeshgangireddy**
## 📝 Description
DAinomaly 2 was released a few months ago. Dinomaly2  extends  Dinomaly with context-Aware Recentering. The other concepts like foundation Transformers, Noisy Bottleneck, Unfocused Linear Attention, Loose Reconstruction,  are identical to what's already implemented. 

Context-Aware Recentering subtracts the CLS (class) token from each patch feature before reconstruction:
This conditions feature reconstruction on class-specific context, preventing the decoder from reconstructing patterns that are "normal" in one object class but anomalous in another.

The code is not public, but for single class UAD scenario, the changes are simple (at least I assume) 

In this PR, added an extra 'use_context_recentering' that toggles between dinomaly2 (if true) and dinomaly implementations. 

Addtionaly added a logic to show the right number of epochs when max_steps is passed (Dinomaly paper uses 5000 steps )

There could be  slighly improvoement (~1 to 2%) in pixel level scores. 
Example numbers for Bottle category MvTecAD with max_steps = 5000

Metric | Dinomaly | Dinomaly2 | Delta
-- | -- | -- | --
image_AUROC | 1 | 1 | 0
image_F1Score | 1 | 0.9839 | -0.0161
pixel_AUROC | 0.9887 | 0.9897 | 0.001
pixel_F1Score | 0.802 | 0.8204 | 0.0184
train_time_s | 5882.9 | 5267.5 | -615.4

Need to test on more categories to get a clear picture. 

## ✨ Changes

Select what type of change your PR is:

- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔄 Refactor (non-breaking change which refactors the code base)
- [ ] ⚡ Performance improvements
- [ ] 🎨 Style changes (code style/formatting)
- [ ] 🧪 Tests (adding/modifying tests)
- [ ] 📚 Documentation update
- [ ] 📦 Build system changes
- [ ] 🚧 CI/CD configuration
- [ ] 🔧 Chore (general maintenance)
- [ ] 🔒 Security update
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [x] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).
- [x] 🏷️ My PR title follows conventional commit format.

For more information about code review checklists, see the [Code Review Checklist](https://github.com/open-edge-platform/anomalib/blob/main/docs/source/markdown/guides/developer/contributing.md).

### CODE REVIEW

**Comment by rajeshgangireddy** [comment]
I ran some experiements on mvtec and visa datasets and compared with dinomaly1, there aren't huge or consistent improvements. But some of the categories get some great improvments. Training time is essentially the same as well. 
So, I suggest merging these changes for now and revisting the implementation once the official code is released. 


% improvement when using dinomaly 2: 
```
100*(D2-D1)/D1 
(D2: Dinomaly2, D1: Dinomaly1)
```

image
image

File: src/anomalib/models/image/dinomaly/lightning_model.py [hunk 1]
```diff
@@ -193,6 +200,59 @@ def __init__(
         self.trainable_modules = torch.nn.ModuleList([self.model.bottleneck, self.model.decoder])
         self._initialize_trainable_modules(self.trainable_modules)
 
+    def on_train_start(self) -> None:
+        """Fix Rich progress bar epoch display when using step-based training.
```
Comment (ashwinvaidya17) [inline src/anomalib/models/image/dinomaly/lightning_model.py]: I feel this isn't limited to just this model. Maybe all models can benefit from this method if they ever use step based training. Should we move this to the base class?

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for adding this. Just one minor comment

File: src/anomalib/models/image/dinomaly/lightning_model.py (hunk 1 above)
Comment (rajeshgangireddy) [reply src/anomalib/models/image/dinomaly/lightning_model.py]: Moved.

File: src/anomalib/callbacks/rich_progress_bar.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Copyright (C) 2025 Intel Corporation
```
Comment (ashwinvaidya17) [inline src/anomalib/callbacks/rich_progress_bar.py]:
```suggestion
# Copyright (C) 2026 Intel Corporation
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for the efforts! One super tiny comment. Can you target `release/lib/v2.4.0` and append an entry in the changelog?
